app/components/upload-csv.js

```diff
@@ -0,0 +1,89 @@
+// https://github.com/EnCiv/undebate-ssp/issues/54
+import React, { useState } from 'react'
+import { createUseStyles } from 'react-jss'
+import cx from 'classnames'
+import FilePlusSvg from '../svgr/file-plus'
+import UploadCSVPopup from './upload-csv-popup'
+
+function UploadCSV(props) {
+    const classes = useStyles()
+    const { className, style, electionOM } = props
+    const [electionObj, electionMethods] = electionOM
+    const [selected, setSelected] = useState(false)
+
+    const handleUploadClick = () => {
+        setSelected(!selected)
+    }
+
+    const closePopup = () => {
+        setSelected(false)
+    }
+
+    return (
+        <div className={cx(className, classes.wrapper)} style={style}>
+            <button
+                type='button'
+                className={cx(
+                    classes.btn,
+                    classes.uploadButton,
+                    selected ? classes.btnSelected : classes.btnDeselected
+                )}
+                onClick={handleUploadClick}
+            >
+                Upload CSV File &nbsp;
+                <FilePlusSvg className={classes.filePlusIcon} />
+            </button>
+            <UploadCSVPopup
+                electionObj={electionObj}
+                electionMethods={electionMethods}
+                closePopup={closePopup}
+                visible={selected}
+                className={classes.popup}
+            />
+        </div>
+    )
+}
+
+export default UploadCSV
+
+const useStyles = createUseStyles(theme => ({
+    wrapper: {
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'flex-start',
+        position: 'relative',
+    },
+    filePlusIcon: {
+        height: theme.iconSize,
+        width: theme.iconSize,
+    },
+    btn: {
+        ...theme.button,
+        display: 'flex',
+        justifyContent: 'space-between',
+        '&:hover': {
+            cursor: 'pointer',
+        },
+    },
+    uploadButton: {
+        width: '13.5625rem',
+        height: '3.375rem',
+        border: `2px solid ${theme.button.borderColor}`,
+    },
+    btnDeselected: {
+        backgroundColor: 'transparent',
+        color: theme.colorSecondary,
+        '& svg path': {
+            stroke: theme.colorSecondary,
+        },
+    },
+    btnSelected: {
+        backgroundColor: theme.colorSecondary,
+        color: 'white',
+    },
+    popup: {
+        position: 'absolute',
+        top: 'calc(100% + 0.75rem)',
+        transform: 'translateX(-25%)',
```

Comment:
I'll try to find a more elegant solution to the hardcode. I believe I have to get the width of either the sub component or the super component, can't remember which, and not sure how easy that will be, but I'll definitely look into it. In terms of the positioning, I was kindof basing this on figma and that this button would only be displayed when there's a side bar as well, so we wouldn't have to worry about the popup getting cutoff, but that's probably not the best assumption. I'll work on something better. And I would assume the messed up padding is from the same issue; I believe that flexbox is trying to shift the file-drop to the right as much as it can, but I'll see if that gets fixed when I fix the cutoff issue.
I'll also add a story to the candidate table similar to the upload csv story to fully test an upload and electionObj update with all of it rendered. 
Thanks @ddfridley for the review, let me know if there's anything else!